src/backend/distributed/citus--7.0-15--7.0-16.sql

```diff
@@ -0,0 +1,24 @@
+/* citus--7.0-15--7.0-16 */
+
+SET search_path = 'pg_catalog';
+
+CREATE OR REPLACE FUNCTION master_move_node(node_name text,
+                                            node_port int,
+                                            new_node_name text,
+                                            new_node_port int)
+RETURNS void
+AS $$
+BEGIN
+  LOCK pg_dist_node IN ACCESS EXCLUSIVE MODE;
+
+  UPDATE pg_dist_node
```

Comment:
Could we write this function in C for easier maintenance? (e.g. if there's a bug we want to backport then this becomes a whole lot easier if we don't need to redefine the function).

Reply:
I probably don't understand what you're saying because I don't see how this makes backports harder. To redefine the function we add an extension update file, that doesn't sound hard, and for something this small SQL is much easier to read and modify.

Reply:
Say we release this function as part of 7.1. We start working on 7.2 and add new functions in citus--7.1-1--7.2-1.sql. Now we discover we really needed to do something extra in master_update_node to prevent problems and decide to backport to 7.1. We define a citus--7.1-1--7.1-2.sql, release a new package, and ask users to run ALTER EXTENSION after the install (otherwise they'll have downtime due to version checks). Now when it comes to upgrading from 7.1 to 7.2 we potentially need to consider two different upgrade paths. One with the patch and one without. We'll need to add an additional citus--7.1-2--7.2-1.sql file to enable upgrading after applying the patch. Every new patch adds yet another upgrade path we need to consider. If someone upgrades without the patch, postgres will search for the shortest path among the upgrade files and will thus pick citus--7.1-1--7.2-1.sql, skipping the patch. So now we have to put the function somewhere else in the upgrade path for those users. Then there are also users who run from master...

If master_update_node was a C function then we could just release a new binary. We'd also make it much easier to find the current version and do version control.

Reply:
If we _really_ want to write functions in PL/pgSQL or SQL, we can always spin up the required machinery and execute the body of our function based on some static string or whatnot. This static string could be updated in a patch update.

But @marcocitus' right otherwise… as much as I want to push for maintainability and using the simplest language for the job, the reality is we don't want to ever have an `ALTER EXTENSION` within a patch update; they're for major updates only; patches should require a simple package update and restart.